modules/ballerina-http/src/main/ballerina/ballerina/net/http/transactions/coordinator/2pc-commons.bal

```diff
@@ -0,0 +1,243 @@
+        boolean prepareDurablesSuccessful = prepare(txn, durableEndpoints);
+        if (prepareDurablesSuccessful) {
+            // If all durable participants voted YES (PREPARED or READONLY), next call notify(commit) on all
+            // (durable & volatile) participants and return committed to the initiator
+            boolean notifyDurablesSuccessful = notify(txn, durableEndpoints, "commit");
+            boolean notifyVolatilesSuccessful = notify(txn, volatileEndpoints, "commit");
+            if (notifyDurablesSuccessful && notifyVolatilesSuccessful) {
+                message = "committed";
+            } else {
+                // return Hazard outcome if a participant cannot successfully end its branch of the transaction
+                err = {msg:"Hazard-Outcome"};
+            }
+        } else {
+            // If some durable participants voted NO, next call notify(abort) on all durable participants
+            // and return aborted to the initiator
+            boolean notifyDurablesSuccessful = notify(txn, durableEndpoints, "abort");
+            boolean notifyVolatilesSuccessful = notify(txn, volatileEndpoints, "abort");
+            if (notifyDurablesSuccessful && notifyVolatilesSuccessful) {
+                if (txn.possibleMixedOutcome) {
+                    message = "mixed";
+                } else {
+                    message = "aborted";
+                }
+            } else {
+                // return Hazard outcome if a participant cannot successfully end its branch of the transaction
+                err = {msg:"Hazard-Outcome"};
+            }
+        }
+    } else {
+        boolean notifySuccessful = notify(txn, volatileEndpoints, "abort");
+        if (notifySuccessful) {
+            if (txn.possibleMixedOutcome) {
+                message = "mixed";
+            } else {
+                message = "aborted";
+            }
+        } else {
+            // return Hazard outcome if a participant cannot successfully end its branch of the transaction
+            err = {msg:"Hazard-Outcome"};
+        }
+    }
+    return;
+}
+
+function notifyAbort (TwoPhaseCommitTransaction txn) returns (string message, error err) {
+    map participants = txn.participants;
+    string transactionId = txn.transactionId;
+    any[] p = participants.values();
+    int i = 0;
+    message = "aborted";
+    while (i < lengthof p) {
+        var participant, _ = (Participant)p[i];
+        Protocol[] protocols = participant.participantProtocols;
+        if (protocols != null) {
+            int j = 0;
+            while (j < lengthof protocols) {
+                Protocol proto = protocols[j];
+                var status, e = notifyParticipant(transactionId, proto.url, "abort");
+                if (e != null) {
+                    err = {msg:"Hazard-Outcome"};
+                    return;
+                } else if(status == "committed") {
```

Comment:
Fix the formatting issues in .bal files. (Space after if etc) Check all the places.